pig/test/org/partiql/pig/domain/TypeDomainSemanticCheckerTests.kt

```diff
@@ -63,13 +63,6 @@ class TypeDomainSemanticCheckerTests {
             TestCase("(define some_domain (domain (sum empty_sum)))",
                      makeErr(1, 29, SemanticErrorContext.EmptySumType("empty_sum"))),
 
-            // Product element arity ordering
-            TestCase("(define some_domain (domain (product some_product x::(? int) y::int)))",
-                     makeErr(1, 62, SemanticErrorContext.RequiredElementAfterOptional)),
-
-            TestCase("(define some_domain (domain (product some_product x::(? int) y::(* int 1))))",
-                     makeErr(1, 62, SemanticErrorContext.ProductCannotHaveBothOptionalAndVariadicElements)),
```

Comment:
So, removing these tests is one thing, however, we need to also add some additional tests to cover the use of these arguments as well.  One way of going about that would be to add types like the ones defined here to `test_domain`  `pig-tests/type-domains/sample-universe.ion` and then add tests in `test/org/partiql/pig/tests/IonElementTransformerTests.kt`